brave/src/main/java/brave/propagation/TraceContextHolder.java

```diff
@@ -0,0 +1,49 @@
+package brave.propagation;
+
+public class TraceContextHolder {
```

Comment:
I did it in this way because I needed a single object which would keep state for each thread. 

I'm wondering how I can achieve similar result using Factory. @devinsba, maybe you have something in mind ?